app/services/LocationService.js

```diff
@@ -80,35 +80,39 @@ export class LocationData {
       }
 
       // Backfill the stationary points, if available
-      // The assumption is that if we see a gap in the data, the
-      // best guess is that the person was in that location for 
-      // the entire time.  This makes it easier for a health authority
+      // The assumption is that if we see a gap in the data, and the 
+      // device hasn't moved significantly, then we can fill in the missing data
+      // with the current location.  This makes it easier for a health authority
       // person to have a set of locations over time, and they can manually
       // redact the time frames that aren't correct.
       if (curated.length >= 1) {
         let lastLocationArray = curated[curated.length - 1];
 
-        // Figure out the time to start the backfill from.  Default is
-        // the time of the last entry in the location set.
-        // To protect ourselves, we won't backfill more than the
-        // maxBackfillTime
-        let lastRecordedTime = lastLocationArray['time'];
-        if ((unixtimeUTC - lastRecordedTime) > this.maxBackfillTime) {
-          lastRecordedTime = unixtimeUTC - this.maxBackfillTime;
-        }
+        var nearby = areLocationsNearby(
+          lastLocationArray['latitude'],
+          lastLocationArray['longitude'],
+          location['latitude'],
+          location['longitude'],
+        );
+        //console.log('[INFO] nearby:', nearby);
 
-        for (
-          let newTS = lastRecordedTime + this.locationInterval;
-          newTS < unixtimeUTC - this.locationInterval;
-          newTS += this.locationInterval
-        ) {
-          let lat_lon_time = {
-            latitude: lastLocationArray['latitude'],
-            longitude: lastLocationArray['longitude'],
-            time: newTS
-          };
-          console.log('[INFO] backfill location:',lat_lon_time);
-          curated.push(lat_lon_time);
+        // Actually do the backfill if the current point is nearby the previous
+        // point and the time is within the maximum time to backfill.
+        let lastRecordedTime = lastLocationArray['time'];
+        if (nearby && unixtimeUTC - lastRecordedTime < this.maxBackfillTime) {
```

Comment:
Code looks good. Tests would be good for this change too.

Reply:
Agreed.  Not ideal, but not adding tests for the LocationService in this PR.  Will address in a followup.